x-pack/legacy/plugins/ml/server/routes/anomaly_detectors.ts

```diff
@@ -0,0 +1,460 @@
+   */
+  router.post(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}/_close',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const options: { jobId: string; force?: boolean } = {
+          jobId: request.params.jobId,
+        };
+        const force = request.query.force;
+        if (force !== undefined) {
+          options.force = force;
+        }
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.closeJob', options);
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {delete} /api/ml/anomaly_detectors/:jobId Delete specified job
+   * @apiName DeleteAnomalyDetectorsJob
+   * @apiDescription Deletes specified anomaly detection job.
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.delete(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const options: { jobId: string; force?: boolean } = {
+          jobId: request.params.jobId,
+        };
+        const force = request.query.force;
+        if (force !== undefined) {
+          options.force = force;
+        }
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.deleteJob', options);
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {post} /api/ml/anomaly_detectors/_validate/detector Validate detector
+   * @apiName ValidateAnomalyDetector
+   * @apiDescription Validates specified detector.
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.post(
+    {
+      path: '/api/ml/anomaly_detectors/_validate/detector',
+      validate: {
+        body: schema.any(),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.validateDetector', {
+          body: request.body,
+        });
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {post} /api/ml/anomaly_detectors/:jobId/_forecast Create forecast for specified job
+   * @apiName ForecastAnomalyDetector
+   * @apiDescription Creates forecast for specified anomaly detection job.
```

Comment:
Edit to  `Creates a forecast for the specified anomaly detection job, predicting the future behavior of a time series by using its historical behavior.`

Reply:
Updated in b70059756cf4d0eb6a125b0ba7e891c69daa622d